app/controllers/application_controller.rb

```diff
@@ -1,14 +1,26 @@
 class ApplicationController < ActionController::Base # :nodoc:
+  respond_to :json
   protect_from_forgery with: :null_session
 
+  include Pundit
+
   before_action :configure_permited_parameters, if: :devise_controller?
 
-  acts_as_token_authentication_handler_for User
+  respond_to :json
+
+  rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+  acts_as_token_authentication_handler_for User, fallback: :none
 
   protected
 
   # allow user to sign up with a name
   def configure_permited_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
   end
+
+  # return status 401 when pundit authorization fails
+  def user_not_authorized
+    render json: nil, status: :unauthorized
```

Comment:
this is the same as nothing: true, or you can also use return head :unauthorized